cs/src/core/Index/CheckpointManagement/DeviceLogCommitCheckpointManager.cs

```diff
@@ -429,6 +372,35 @@ public void OnRecovery(Guid indexToken, Guid logToken)
             }
         }
 
+        /// <inheritdoc />
+        public void OnRecovery(long commitNum, bool purgeEarlierCommits)
+        {
+            if (removeOutdated || purgeEarlierCommits)
```

Comment:
This block is slightly confusing to me with many if conditionals: maybe just flatten it into:
```
if (removeOutdated) {
  ... 
}
else if (purgeEarlierCommits) {
 ...
return;
}
```

Reply:
fixed